### PR TITLE
[scroll-animations-1] Add self keyword

### LINKED ISSUE #8227: [scroll-animations-1] Allow Anonymous Scroll Progress Timelines to target self
When creating an Anonymous Scroll Progress Timelines usning `scroll()` you can target either the `root` or the `nearest` scroller. 

Currently `nearest` is defined as:

> Specifies to use the nearest ancestor scroll container. _(Default.)_

This does not allow authors to create an Anonymous Scroll Progress Timeline that uses the element itself, as `nearest` only looks to ancestors. E.g. the code below does not work when `.container` is the scroller

```css
.container {
  animation: 1s bg-scale both linear;
  animation-timeline: scroll(nearest inline);
}
```

As a result, authors must use a named Scroll Progress Timeline instead:

```css
.container {
  animation: 1s bg-scale both linear;
  scroll-timeline: --self inline;
  animation-timeline: --self;
}
```

This seems like a bit a hassle. Can we adjust `nearest` so that it starts looking from the targeted element itself first, and only then to ancestors? Or maybe introduce a new keyword `self` should altering `nearest` introduce some nasty side-effects?

### PR DESCRIPTION
**Pull request by bramus**
Fixes #8227.

This PR replaces #8390 which was created from the wrong branch.

### CODE REVIEW

**Comment by bramus** [comment]
Thanks for the feedback @fantasai; greatly appreciate it. Did some changes which should address your remarks. Happy to take more pointers, as this is all fairly new to me.

**Comment by fantasai** [comment]
Looks pretty good! The one other thing to fix is the line breaks.
Read https://rhodesmill.org/brandon/2012/one-sentence-per-line/ ; then take a look at the rest of the document to get a sense of how we're formatting the source.

**Comment by bramus** [comment]
Interesting link. Makes sense, instead of cutting off mid-sentence when the line length limit is almost reached. Think I got it right now.

**Comment by flackr** [comment]
> You need to define what happens when the element's own box isn't a [=scroll container=].

I would have thought these cases are covered by the text in the [ScrollTimeline interface](https://drafts.csswg.org/scroll-animations-1/#scrolltimeline-interface):
> If the [source](https://drafts.csswg.org/scroll-animations-1/#dom-scrolltimeline-source) of a [ScrollTimeline](https://drafts.csswg.org/scroll-animations-1/#scrolltimeline) is an element whose [principal box](https://drafts.csswg.org/css-display-4/#principal-box) does not exist or is not a [scroll container](https://drafts.csswg.org/css-overflow-3/#scroll-container), or if there is no [scrollable overflow](https://drafts.csswg.org/css-overflow-3/#scrollable-overflow), then the [ScrollTimeline](https://drafts.csswg.org/scroll-animations-1/#scrolltimeline) is [inactive](https://drafts.csswg.org/web-animations-1/#inactive-timeline).

Do we define this for [scroll-timeline-name](https://drafts.csswg.org/scroll-animations-1/#scroll-timeline-name) as well since that can target a non scroll container?

File: scroll-animations-1/Overview.bs [hunk 1]
```diff
@@ -181,6 +181,13 @@ spec: cssom-view-1; type: dfn;
 		<dt><dfn>root</dfn>
 		<dd>
 			Specifies to use the document viewport as the [=scroll container=].
+
+		<dt><dfn>self</dfn>
+		<dd>
+			Specifies to use the element’s own [=principal box=] as the [=scroll container=].
+			If the [=principal box=] is not a [=scroll container=], 
+			or if there is no [=scrollable overflow=],
```
Comment (flackr) [inline scroll-animations-1/Overview.bs:189]: I think this condition at least should already be covered by the definition for [Scroll Progress Timelines](https://drafts.csswg.org/scroll-animations-1/#scroll-timelines):

> If the 0% position and 100% position coincide (i.e. the scroll container has no overflow to scroll), the timeline is [inactive](https://drafts.csswg.org/web-animations-1/#inactive-timeline).

**Review by flackr — APPROVED** [review]

File: scroll-animations-1/Overview.bs (hunk 1 above)
Comment (fantasai) [reply scroll-animations-1/Overview.bs:192]: I think you're right, and this line should be removed.

```suggestion
```

**Comment by bramus** [comment]
@fantasai I’ve committed the suggested change.

**Review by fantasai — APPROVED** [review]

**Pull request merged**